analytics-spring-boot-starter/src/main/java/com/segment/analytics/autoconfigure/SegmentAnalyticsAutoConfiguration.java

```diff
@@ -20,7 +21,9 @@ public class SegmentAnalyticsAutoConfiguration {
   @Autowired private SegmentProperties properties;
 
   @Bean
-  public Analytics segmentAnalytics() {
-    return Analytics.builder(properties.getWriteKey()).build();
+  public Analytics segmentAnalytics(ObjectProvider<SegmentAnalyticsCustomizer> customizerProvider) {
+    Analytics.Builder builder = Analytics.builder(properties.getWriteKey());
+    customizerProvider.orderedStream().forEach((customizer) -> customizer.customize(builder));
+    return builder.build();
```

Comment:
This implementation was inspired by Spring Boot's autoconfiguration for the webclient;

https://github.com/spring-projects/spring-boot/blob/d55cd3b46fc33e9c2806fac0079993954c62a7d3/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/function/client/WebClientAutoConfiguration.java#L54-L56